controller/forwarding/hyperlane_test.go

```diff
@@ -0,0 +1,475 @@
+			handler := mocks.HyperlaneHandler{Tokens: make(map[string]warptypes.WrappedHypToken)}
+
+			if tC.setup != nil {
+				tC.setup(&handler)
+			}
+
+			controller, err := forwarding.NewHyperlaneController(
+				log.NewNopLogger(),
+				handler,
+			)
+			require.NoError(t, err)
+
+			err = controller.ValidateForwarding(
+				context.Background(),
+				tC.transferAttr,
+				tC.hypAttr,
+			)
+
+			if tC.expError != "" {
+				require.ErrorContains(t, err, tC.expError)
+			} else {
+				require.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestHandlePacket_Hyperlane(t *testing.T) {
+	usdnID := make([]byte, 32)
+	copy(usdnID, "usdn id")
+
+	testCases := []struct {
+		name     string
+		setup    func(*mocks.HyperlaneHandler) context.Context
+		packet   func() *types.ForwardingPacket
+		expError string
+	}{
+		{
+			name:     "error - when the attributes are valid",
+			packet:   func() *types.ForwardingPacket { return nil },
+			expError: "Hyperlane controller received nil packet",
+		},
+		{
+			name: "error - when forwarding is default values",
+			packet: func() *types.ForwardingPacket {
+				a, err := types.NewTransferAttributes(
+					core.PROTOCOL_IBC,
+					"channel-0",
+					"usdn",
+					math.NewInt(1),
+				)
+				require.NoError(t, err)
+
+				return &types.ForwardingPacket{
+					TransferAttributes: a,
+					Forwarding:         &core.Forwarding{},
+				}
+			},
+			expError: "error extracting Hyperlane forwarding",
+		},
+		{
+			name: "error - when transfer attributes are default values",
+			packet: func() *types.ForwardingPacket {
+				f, err := forwardingtypes.NewHyperlaneForwarding(
+					usdnID,
+					0,
+					make([]byte, 32),
+					make([]byte, 32),
+					"",
+					math.ZeroInt(),
+					sdk.NewCoin("usdn", math.ZeroInt()),
+					[]byte{},
+				)
+				require.NoError(t, err)
+
+				return &types.ForwardingPacket{
+					TransferAttributes: &types.TransferAttributes{},
+					Forwarding:         f,
+				}
+			},
+			expError: "error validating Hyperlane forwarding",
+		},
+		{
+			name: "error - when the remote transfer fails",
+			setup: func(m *mocks.HyperlaneHandler) context.Context {
+				tokenID := hyperlaneutil.HexAddress(usdnID)
+				hypToken := &warptypes.WrappedHypToken{
+					Id:            tokenID.String(),
+					Owner:         "noble",
+					TokenType:     0,
+					OriginMailbox: "",
+					OriginDenom:   "usdn",
+					IsmId:         &hyperlaneutil.HexAddress{},
+				}
+
+				m.Tokens[hypToken.Id] = *hypToken
+
+				return context.WithValue(context.Background(), mocks.FailingContextKey, true)
+			},
+			packet: func() *types.ForwardingPacket {
+				a, err := types.NewTransferAttributes(
+					core.PROTOCOL_IBC,
+					"channel-0",
+					"usdn",
+					math.NewInt(1),
+				)
+				require.NoError(t, err)
+
+				f, err := forwardingtypes.NewHyperlaneForwarding(
+					usdnID,
+					0,
+					make([]byte, 32),
+					make([]byte, 32),
+					"",
+					math.ZeroInt(),
+					sdk.NewCoin("usdn", math.ZeroInt()),
+					[]byte{},
+				)
```

Comment:
also here it could be very helpful to refactor to only adjust the fields that are required to be wrong, based on a given `validHyperlaneForwardingPacket` :pray: